### PR TITLE
OCPBUGS-39035: Ensure AzureCluster.Status.FailureDomains is populated when any NodePool specifies an Availability Zone

### PR DESCRIPTION
**Pull request by fxierh**
Ensure AzureCluster.Status.FailureDomains is populated when any NodePool specifies an Availability Zone.

Otherwise CAPZ will create Azure VMs with both Availability Zone and Availability Set configured, which results in an error:
```bash
{
        "error": {
          "code": "BadRequest",
          "message": "Virtual Machine cannot be created because both Availability Zone and Availability Set were specified. Deploying an Availability Set to an Availability Zone isn’t supported."
        }
}
```

ref: 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/711d86187262612c8f7f02f1848c22fbf3b040cd/azure/scope/cluster.go#L778-L780

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go [hunk 1]
```diff
@@ -242,6 +242,22 @@ func reconcileAzureCluster(azureCluster *capiazure.AzureCluster, hcluster *hyper
...
+	// (which results in an error), ensure AzureCluster.Status.FailureDomains is populated when any NodePool
+	// specifies an Availability Zone.
+	npList, err := listNodePools(ctx, c, hcluster.Namespace, hcluster.Name)
+	if err != nil {
+		return err
+	}
+	for _, np := range npList {
+		if len(np.Spec.Platform.Azure.AvailabilityZone) > 0 {
+			azureCluster.Status.FailureDomains = capiv1.FailureDomains{
+				"1": capiv1.FailureDomainSpec{},
```
Comment (Patryk-Stefanski) [inline hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go:255]: would we not want a failureDomain per AZ ?

**Comment by fxierh** [comment]
Fixed upstream: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5098

/close

**Pull request closed without merging**